File: File Reading Examples/Python API/File_Reader.py
from DelsysFile import DelsysFile

class File_Reader:
    """File Reader Interface to FileReaderAPI"""

    def __init__(self):
        self.filePath = ""
        self.reader = []
        self.trials = []

    def readFile(self, filePath):
        """Pass path to .shpf file - FileReaderAPI will parse the file"""
        import Delsys.FileManager.Reader as FM
        self.filePath =  filePath
        self.reader = FM.DelsysFileReader(filePath)

    def ParsedFile(self) -> DelsysFile:
        """Once the file is parsed return the DelsysFile object"""
        self.trials = list(self.reader.OpenFileArray())
        return self.Trial(0)

    def Trial(self, selected_trial: int) -> DelsysFile:
        """Trial for given index"""
        return DelsysFile(self.trials[selected_trial])

    def TrialCount(self) -> int:
        """Returns the amount of trials in the file"""
        return len(self.trials)

    def Close(self):
        """Close the file"""
        self.reader.Close()

    def FileType(self) -> str:
        """Returns the file type or extension"""

        readType = int(self.reader.FileType)
        if readType == 1:
            fileType = '.shpf'
        elif readType == 5:
            fileType = '.delsys'
        else:
            fileType = 'Not a .shpf'
        return fileType

File: File Reading Examples/MATLAB API/FileReader.m
classdef FileReader < handle
    %File Reader Interface to FileManagerAPI% 
    
    properties (Access = private)
       pathToDll;
       reader;
       file;
       trials;
       
    end
    
    methods  
        function obj = FileReader(pathToDll)
        %Pass path dll dependencies.% 
            dotnetenv("core");
            obj.pathToDll = pathToDll;
            NET.addAssembly(pathToDll + "Delsys.FileReader.dll");
            NET.addAssembly(pathToDll + "Delsys.FileManager.dll");
        end
        
        function Read(obj,filePath)
        %Pass path to .shpf file - FileReaderAPI will parse the file% 
            obj.reader = Delsys.FileManager.Reader.DelsysFileReader(filePath);
        end
        
        function file = ParsedFile(obj)
        %Once the file is paresd return the DelsysFile object% 
          obj.trials = obj.reader.OpenFileArray();
          file = DelsysFile(obj.trials(1));
        end

        function count = TrialCount(obj)
        %Number of trials in the file% 
          count = obj.trials.Length;
        end

        function file = Trial(obj, selectedTrial)
        %Get Trial at an index% 
          file = DelsysFile(obj.trials(selectedTrial));
        end
        
        function fileType = FileType(obj)
        %Returns the file type or extension% 
        
          fileType = string(obj.reader.FileType);
        end

        function Close(obj)
        %Closes the file type or extension% 
        
          obj.reader.Close();
        end
        
    end
    
end

